sdk/samples/iot/paho_iot_pnp_component_sample.c

```diff
@@ -0,0 +1,626 @@
+    char* topic,
+    int topic_len,
+    MQTTClient_message const* receive_message)
+{
+  az_result rc;
+
+  az_span const topic_span = az_span_create((uint8_t*)topic, topic_len);
+  az_span const message_span
+      = az_span_create((uint8_t*)receive_message->payload, receive_message->payloadlen);
+
+  az_iot_hub_client_properties_response property_response;
+  az_iot_hub_client_command_request command_request;
+
+  // Parse the incoming message topic and handle appropriately.
+  rc = az_iot_hub_client_properties_parse_received_topic(
+      &hub_client, topic_span, &property_response);
+  if (az_result_succeeded(rc))
+  {
+    IOT_SAMPLE_LOG_SUCCESS("Client received a valid property topic response.");
+    IOT_SAMPLE_LOG_AZ_SPAN("Topic:", topic_span);
+    IOT_SAMPLE_LOG_AZ_SPAN("Payload:", message_span);
+    IOT_SAMPLE_LOG("Status: %d", property_response.status);
+
+    handle_device_property_message(receive_message, &property_response);
+  }
+  else
+  {
+    rc = az_iot_hub_client_commands_parse_received_topic(&hub_client, topic_span, &command_request);
+    if (az_result_succeeded(rc))
+    {
+      IOT_SAMPLE_LOG_SUCCESS("Client received a valid command topic response.");
+      IOT_SAMPLE_LOG_AZ_SPAN("Topic:", topic_span);
+      IOT_SAMPLE_LOG_AZ_SPAN("Payload:", message_span);
+
+      handle_command_request(receive_message, &command_request);
+    }
+    else
+    {
+      IOT_SAMPLE_LOG_ERROR("Message from unknown topic: az_result return code 0x%08x.", rc);
+      IOT_SAMPLE_LOG_AZ_SPAN("Topic:", topic_span);
+      exit(rc);
+    }
+  }
+}
+
+// process_device_property_message handles incoming properties from Azure IoT Hub.
+static void process_device_property_message(
+    az_span property_message_span,
+    az_iot_hub_client_properties_response_type response_type)
+{
+  az_result rc = az_iot_hub_client_properties_patch_get_publish_topic(
+      &hub_client,
+      pnp_mqtt_get_request_id(),
+      publish_message.topic,
+      publish_message.topic_length,
+      NULL);
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(rc, "Failed to get the property PATCH topic");
+
+  az_json_reader jr;
+  az_span component_name;
+  int32_t version = 0;
+  rc = az_json_reader_init(&jr, property_message_span, NULL);
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(rc, "Could not initialize the json reader");
+
+  rc = az_iot_hub_client_properties_get_properties_version(
+      &hub_client, &jr, response_type, &version);
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(rc, "Could not get the property version");
+
+  rc = az_json_reader_init(&jr, property_message_span, NULL);
+  IOT_SAMPLE_EXIT_IF_AZ_FAILED(rc, "Could not initialize the json reader");
+
+  // Applications call az_iot_hub_client_properties_get_next_component_property to enumerate
+  // properties received.
+  while (az_result_succeeded(
+      rc = az_iot_hub_client_properties_get_next_component_property(
+          &hub_client, &jr, response_type, AZ_IOT_HUB_CLIENT_PROPERTY_WRITEABLE, &component_name)))
+  {
+    if (rc == AZ_OK)
```

Comment:
is this check required?

Reply:
Yes - in `az_iot_hub_client_properties_get_next_component_property` there's checks where we can return a non AZ_OK.